ocaml/xapi/xapi_xenops.ml

```diff
@@ -2389,7 +2391,6 @@ let maybe_cleanup_vm ~__context ~self =
 		 * will be called with events enabled and therefore we get Xenopsd into a
 		 * consistent state with Xapi *)
 		Events.with_suppressed __context self queue_name dbg id (fun _ -> ());
-		Xenopsd_metadata.delete ~__context id;
```

Comment:
This line makes me nervous. On discussion we could make Xenopsd_metadata.delete only remove the cache if the XenopsAPI.VM.remove succeeds as I think we're relying on that failing for VMs that are actually running.